### ISSUE TITLE
qwen1.5的sft数据处理部分使用了预训练的数据处理方式

### ISSUE THREAD

**Issue by bjr-yql** [QwenLM/Qwen1.5#396]
for i, msg in enumerate(messages):
        texts.append(
            tokenizer.apply_chat_template(
                msg,
                chat_template=TEMPLATE,
                tokenize=True,
                add_generation_prompt=False,
                padding=True,
                max_length=max_len,
                truncation=True,
            )
        )
    input_ids = torch.tensor(texts, dtype=torch.int)
    target_ids = input_ids.clone()
    target_ids[target_ids == tokenizer.pad_token_id] = IGNORE_TOKEN_ID
    attention_mask = input_ids.ne(tokenizer.pad_token_id)

    return dict(
        input_ids=input_ids, target_ids=target_ids, attention_mask=attention_mask
    )

**Comment by jklj077** [comment]
Closed as duplicate. See <https://github.com/QwenLM/Qwen1.5/issues/391> for an index.
